full version of programm (#1)
oleh-mazurkov/cinema-app: README.md

# cinema-app

> **What this application is for and what it can do**

Reviewer: ```suggestion
**What this application is for?**
```

> Primitive demonstration program of service of cinema services and its clients

Reviewer: ```suggestion
Basic implementation of a cinema with all its functions.
```

> **Technologies used**

Reviewer: ```suggestion
**Technologies**
```

Program based on the use of:
- Hibernate 

> - Spring framework

Reviewer: - Java 11
- MySQL
- Hibernate
- Spring (Core, WEB, Security)
- Apache Tomcat (to run app locally)

**How to run a project**

> Before lunching the programm you need to install on you comp:

Reviewer: ```suggestion
Before launching the program you need to install:
```

> MySQL Workbench (DB for datas working) and Apache Tomcat (to lunch the programm)

Reviewer: make a list of these two

To lunch the project you need to copy program **cinema-app-1.0-SNAPSHOT.war** from this server
and place it in /webapps Tomcat-programm`s on your comp.
Lunch **/bin/startup.bat** Tomcat-programm/
In brouser you can see authentication data to start of the program.
But at first you need to inject previus data in your DB for first authentication.
So you need to lunch in brouser: **http://localhost:8080/inject**
You can see "Done!" in brousers window (or "HTTP Status 500 – Internal Server Error" - dont pay your attantion on it)).
Next step - authentication: lunch in brouser **http://localhost:8080/login** 

> and input "alice@i.ua" as login, "12345678" as password (this person have role "ADMIN")

Reviewer: It should look like numbered list and much shorter, you should remove target folder, so you don't need to mention it here

   1.2 Add movies to the database (with movie title and description) and get a list of all movies
   1.3 Add, edit and delete sessions (including: Cinema, Movie, show date and time),
   and get a list of sessions by movie title and show date
2. Customer services
   The program allows:
   2.1 Register a client (includes: email and password) as USER - external cinema client
   with automatic creation of the client's card
   `ADMINs (which will be able to use, enter and edit data on cinema services)
   entered into InjectController programs and, when creating a database, are entered with the command /inject`
   2.2 Authenticate the client when logging in by his email and password
   2.3 Ticket formation (includes: session and client)
   2.4 Forming a customer card (includes: tickets and customer)
   2.5 Formation of the order with display of the current date of the order, tickets and the client

Function check:
1 Running the program via Tomcat (an empty database "ticket_app" is created in MySQL)
2 Filling the database with test data through a browser command: **http://localhost:8080/inject**

> 3 Login via **http://localhost:8080/login** "alice@i.ua" password "12345678" (role "ADMIN")

Reviewer: add section Features and describe here  all available endpoints in form of list where information about endpoint should be like this:
GET: /cinema-halls - display a list of cinema halls (admin or user role)
